src/segment_memory.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"strconv"
+
+	mem "github.com/pbnjay/memory"
+)
+
+type memory struct {
+	props       *properties
+	env         environmentInfo
+	TotalMemory uint64
+	FreeMemory  uint64
+}
+
+const (
+	Precision Property = "precision"
+)
+
+func (n *memory) enabled() bool {
+	return true
+}
+
+func (n *memory) string() string {
+	newText := strconv.FormatFloat(100.0/float64(n.TotalMemory)*float64(n.TotalMemory-n.FreeMemory), 'f', n.props.getInt(Precision, 0), 64)
```

Comment:
Can we split the calculation and formatting for readability? Other than this 😗👌🏻